compass/ocean/tests/isomip_plus/initial_state.py

```diff
@@ -149,44 +189,52 @@ def _compute_initial_condition(self):
         min_depth = -ds.ssh + min_column_thickness
         ds['bottomDepth'] = np.maximum(ds.bottomDepth, min_depth)
         print(f'Adjusted bottomDepth for '
-              f'{np.sum(ds.bottomDepth.values<min_depth.values)} cells '
+              f'{np.sum(ds.bottomDepth.values < min_depth.values)} cells '
               f'to achieve minimum column thickness of {min_column_thickness}')
 
+        # Initialize vertical coordinate without time dimension
+        ds = ds.isel(Time=0)
         init_vertical_coord(config, ds)
 
-        max_bottom_depth = -config.getfloat('vertical_grid', 'bottom_depth')
-        frac = (0. - ds.zMid) / (0. - max_bottom_depth)
-
         # compute T, S
         init_top_temp = section.getfloat('init_top_temp')
         init_bot_temp = section.getfloat('init_bot_temp')
         init_top_sal = section.getfloat('init_top_sal')
         init_bot_sal = section.getfloat('init_bot_sal')
-        thin_film_mask = np.logical_and(mask.values,
-                                        np.logical_not(floating_mask.values))
-        # These coefficients are hard-coded as the defaults in the namelist
-        # Note that using the land ice pressure rather than the pressure at
-        # floatation will mean that there is a small amount of cooling from
-        # grounding line retreat. However, the thin film should be thin enough
-        # that this effect isn't signicant.
-        freezing_temp = (6.22e-2 +
-                         -5.63e-2 * init_bot_sal +
-                         -7.43e-8 * landIcePressure +
-                         -1.74e-10 * landIcePressure * init_bot_sal)
-        _, thin_film_temp = np.meshgrid(ds.refZMid, freezing_temp)
-        _, thin_film_mask = np.meshgrid(ds.refZMid, thin_film_mask)
-        thin_film_temp = np.expand_dims(thin_film_temp, axis=0)
+
         if self.vertical_coordinate == 'single_layer':
-            ds['temperature'] = init_bot_temp * xr.ones_like(frac)
-            ds['salinity'] = init_bot_sal * xr.ones_like(frac)
+            # Initialize constant T,S
+            ds['temperature'] = init_bot_temp * xr.ones_like(ds.zmid)
+            ds['salinity'] = init_bot_sal * xr.ones_like(ds.zmid)
```

Comment:
I'm seeing errors in my testing.  This needs to be:
```suggestion
            ds['temperature'] = init_bot_temp * xr.ones_like(ds.zMid)
            ds['salinity'] = init_bot_sal * xr.ones_like(ds.zMid)
```

Reply:
`frac` also needs to be defined here, since it's returned below.

Reply:
I would change this whole if/else as follows:
``` python
        if self.vertical_coordinate == 'single_layer':
            # Initialize constant T,S with bottom values
            frac = xr.ones_like(ds.zMid)
        else:
            # Initialize T,S as linear functions with max depth
            max_bottom_depth = -config.getfloat('vertical_grid',
                                                'bottom_depth')
            frac = (0. - ds.zMid) / (0. - max_bottom_depth)
        ds['temperature'] = \
            (1.0 - frac) * init_top_temp + frac * init_bot_temp
        ds['salinity'] = \
            (1.0 - frac) * init_top_sal + frac * init_bot_sal
```

Reply:
Resolved with removal of single layer cases